lib/graphlib/graphlib_graph.ml

```diff
@@ -1094,28 +1094,39 @@ module Fixpoint = struct
   type ('n,'d) t = Solution : {
       steps : int option;
       iters : int;
-      init : 'd;
+      default : 'd;
       approx : ('n,'d,_) Map.t;
     } -> ('n,'d) t
 
 
-  let create constraints init = Solution {
+  let create constraints default = Solution {
       steps=Some 0; iters=0;
       approx=constraints;
-      init;
+      default;
     }
 
   let iterations (Solution {iters}) = iters
 
-  let get (Solution {approx; init}) n =
+  let default (Solution {default}) = default
+
+  let get (Solution {approx; default}) n =
     match Map.find approx n with
-    | None -> init
+    | None -> default
     | Some x -> x
 
   let is_fixpoint (Solution {steps; iters}) = match steps with
     | None -> iters > 0
     | Some steps -> iters < steps
 
+  let derive (Solution s) ~f default =
+    let f ~key ~data = f key data in
+    Solution {
```

Comment:
We need to reset the step and iter counters, since this is a fresh solution, though based on an existing one. You may reuse the create function.

Reply:
Got it. I wasn't sure which way it should go.